src/renderer/containers/Settings/Diagnostic.tsx

```diff
@@ -0,0 +1,343 @@
+  const [cgnatCommandCopied, setCgnatCommandCopied] = React.useState(false);
+  const onCgnatCommandCopy = React.useCallback(() => {
+    window.electron.clipboard.writeText(cgnatCommand);
+    setCgnatCommandCopied(true);
+    window.setTimeout(() => setCgnatCommandCopied(false), 2000);
+  }, [cgnatCommand]);
+  const getCgnatSection = () => {
+    if (cgnat === Presence.UNKNOWN) {
+      return (
+        <>
+          <Typography variant="subtitle2">{cgnatTitle}</Typography>
+          <LoadingButton loading={true} />
+        </>
+      );
+    }
+    if (cgnat === Presence.FAILED && !ipAddress) {
+      return <></>;
+    }
+    return (
+      <>
+        <Typography variant="subtitle2">{cgnatTitle}</Typography>
+        <DialogBody>{cgnatDescription}</DialogBody>
+      </>
+    );
+  };
+
+  const getCgnatCommandSection = () => {
+    if (cgnat === Presence.FAILED && ipAddress) {
+      return (
+        <>
+          <Typography variant="subtitle2">{"Run this command"}</Typography>
+          <AlignCenterDiv>
+            <InputBase css={inputBaseCss} disabled={true} value={displayedCgnatCommand} />
+            <Button variant="contained" color="secondary" onClick={onCgnatCommandCopy} style={buttonStyle}>
+              {cgnatCommandCopied ? "Copied!" : "Copy"}
+            </Button>
+          </AlignCenterDiv>
+          <DialogBody>
+            {"More than one hop to your external IP address indicates CGNAT or Double NAT (or VPN)."}
+          </DialogBody>
+        </>
+      );
+    }
+    return <></>;
+  };
+
+  const [diagnosticResultsCopied, setDiagnosticResultsCopied] = React.useState(false);
+  const onDiagnosticResultsCopy = React.useCallback(() => {
+    let diagnosticResults = `${natTypeTitle}\n${natTypeDescription}\n\n${portMappingTitle}\n${portMappingDescription}`;
+    if (ipAddress) {
+      diagnosticResults += `\n\n${cgnatTitle}\n${cgnatDescription}`;
+    }
+    window.electron.clipboard.writeText(diagnosticResults);
+    setDiagnosticResultsCopied(true);
+    window.setTimeout(() => setDiagnosticResultsCopied(false), 2000);
+  }, [
+    cgnatDescription,
+    cgnatTitle,
+    ipAddress,
+    natTypeDescription,
+    natTypeTitle,
+    portMappingDescription,
+    portMappingTitle,
+  ]);
+
+  return (
+    <>
+      <ActionButton
+        startIcon={<NetworkCheckIcon fill={colors.purpleLighter} style={{ height: 18, width: 18 }} />}
+        color="secondary"
+        variant="contained"
+        onClick={openDialog}
+      >
+        Check network issues
+      </ActionButton>
+      <Dialog open={dialogOpen} closeAfterTransition={true} onClose={() => setDialogOpen(false)} fullWidth={true}>
+        <DialogTitle>Network Diagnostic</DialogTitle>
+        <DialogContent>
+          {getNatTypeSection()}
+          {getPortMappingSection()}
+          {getCgnatSection()}
+          {getCgnatCommandSection()}
+        </DialogContent>
```

Comment:
```suggestion
        <DialogContent>
          <NatTypeSection natType={natType} />
          <PortMappingSection portMapping={portMapping} />
          <CgnatSection cgnat={cgnat} />
          <CgnatCommandSection whateverPropsAreNecessaryToRender={stuff} />
        </DialogContent>
```

This is a React component so prefer to compose individual components into one rather than having all these functions and strings everywhere.